### ISSUE TITLE
Less strict checks on remote files

### ISSUE THREAD

**Issue by dara2** [mjordan/islandora_workbench#620]
When running the --check, if Workbench comes across a URL in the "file" column that it can't access due to something other than a 403 error, it throws an error and I have to start the --check all over again. In the case of files on the server, Workbench will just flag those errors in the workbench.log file and keep moving, as long as you have `strict_check: false` in your config file. It would be great if the same thing could apply to remote files not just for 403's but for 404s and file extension errors too.
Example errors during --check process where Workbench stops completely:
```
28-Apr-23 14:55:03 - ERROR - Remote file "https://therevolutionarycity.org/islandora/object/apsrevcity%3A142879/datastream/JP2/view" identified in CSV "file" column for record with ID "apsrevcity:14279" not found or not accessible (HTTP response code 404).
```
```
28-Apr-23 15:10:50 - ERROR - File "https://therevolutionarycity.org/islandora/object/apsrevcity%3A142879/datastream/JP2/view" in CSV row "apsrevcity:14279" has an extension (html) that is not allowed in the "field_media_file" field of the "file" media type.
```

The one type of error for remote files where Workbench WILL keep going and just record the error in the log is a 403 error, like this:
```
11-May-23 16:25:47 - ERROR - Remote file "https://therevolutionarycity.org/islandora/object/lcprevcity:282/datastream/JP2/view" identified in CSV "file" column for record with ID "lcprevcity:282" not found or not accessible (HTTP response code 403).
```

Noah made a small edit locally to change the 404 and file extension errors to warnings, and can submit a PR if you'd like?

**Comment by noahwsmith** [comment]
@mjordan how would you feel if I were to introduce a new variable like `perform_soft_checks` which, if set to true, would essentially skip the relevant `sys.exit('Error: ' + message)` calls within the `check` routine?

Here are the ones we're skipping in our work to make `check`s more efficient:
* Parent/child order issues https://github.com/mjordan/islandora_workbench/blob/2373149021d1f31f175e5a0090f03aa353074f09/workbench_utils.py#L4943C23-L4943C23
* File extension checks, since in our experience 404s are often expressed as a file extension mismatch https://github.com/mjordan/islandora_workbench/blob/2373149021d1f31f175e5a0090f03aa353074f09/workbench_utils.py#L2012
* EDTF dates validation, since we want a whole list of them for a file, not to fail on every one https://github.com/mjordan/islandora_workbench/blob/2373149021d1f31f175e5a0090f03aa353074f09/workbench_utils.py#L4855

Right now I'm just doing this pretty crudely by commenting that `exit` line out, e.g. 
https://github.com/Born-Digital-US/islandora_workbench/commit/ce2cf4fbb6fdee186434326fca43f996b0fec9bf

But doing this properly would give us a mechanism to keep adding `soft_check` exclusions in the future too.

**Comment by mjordan** [comment]
Hi @noahwsmith this is essentially what #268 was intended to do. I have been thinking about addressing this issue for some time and the approach you are suggesting is probably the best/simplest way to do it. 

I've already got a `strict_check` config setting which defaults to `true`. Making Workbench not die on the first error would change its current behavior a lot, but it probably would be a change most users would welcome. If we use `perform_soft_checks `, the default would be 'true' and those who want it to die (e.g. if it is used in automated ingests) would have to set `perform_soft_checks ` to 'false`. If we use  `strict_check` the default would be `false` and it would need to be set to `true` to not die.

It would be awesome if you could do this work. I very much prefer making the changes in small batches as you suggest above, rather than *all* checks in one big PR. I also think we need to consult with the user community about changing Workbench's current behavior, and how to retain that behavior via whatever config setting we settle on. Would you like to open the conversation in Slack or would you like me to?

**Comment by noahwsmith** [comment]
@mjordan I'd be happy to. Let me know how this looks? ^

I implemented `perform_soft_checks` as described but with the default as `true`, and I refactored the `strict_check` calls by flipping the logic to use the new variable.

**Comment by mjordan** [comment]
Resolved with dfa60ff97b264cf0e4bae81ae24b02215766c6ef.

**Comment by mjordan** [comment]
I've documented the new `perform_soft_checks` at https://mjordan.github.io/islandora_workbench_docs/configuration/#miscellaneous-settings.
